### ISSUE TITLE
💻 Cannot upload macOS application to App Store Connect when using HeapSwiftCore v0.3.0

### ISSUE THREAD

**Issue by AndrewVebster** [heap/heap-swift-core-sdk#2]
Hey 👋 

Our team is migrating from https://github.com/heap/heap-ios-sdk.git to this new `HeapSwiftCore` SDK in hopes to get native support for our macOS application. But we stumbled upon this issue 👇.

## Preconditions
1. `HeapSwiftCore` linked as dependency using Swift Package Manager: `.package(url: "https://github.com/heap/heap-swift-core-sdk.git", .upToNextMinor(from: "0.3.0"))` 
2. It's native macOS application
3. Xcode 14.2

## Steps to reproduce
1. Archive the application
2. Try to upload it to App Store Connect

## Result
```
error: exportArchive: /Users/vagrant/Library/Developer/Xcode/Archives/2023-07-03/XXX 2023-07-03 11.46.02.xcarchive/Products contains invalid products.
Error Domain=IDEFoundationErrorDomain Code=1 "/Users/vagrant/Library/Developer/Xcode/Archives/2023-07-03/XXX 2023-07-03 11.46.02.xcarchive/Products contains invalid products." UserInfo={NSLocalizedDescription=/Users/vagrant/Library/Developer/Xcode/Archives/2023-07-03/XXX 2023-07-03 11.46.02.xcarchive/Products contains invalid products., NSLocalizedRecoverySuggestion=Multiple binaries share the same codesign path:
/Applications/Toggl Track.app/Contents/Frameworks/HeapSwiftCoreInterfaces.framework/Versions/Current
Binaries:
/Applications/Toggl Track.app/Contents/Frameworks/HeapSwiftCoreInterfaces.framework/Versions/Current/HeapSwiftCoreInterfaces
/Applications/Toggl Track.app/Contents/Frameworks/HeapSwiftCoreInterfaces.framework/HeapSwiftCoreInterfaces
This can happen if your build process copies frameworks by following symlinks. }
```

## Other notes
1. iOS app can be uploaded successfully
2. Maybe it's related to this Xcode issue: https://forums.swift.org/t/swiftpm-binarytarget-dependency-and-code-signing/38953
3. macOS app can be uploaded when using this specific revision: `.package(url: "https://github.com/heap/heap-swift-core-sdk.git", .revision("9aebbae70a1d05fa58a1c02655ae5208a0e4bf22"))`. But this means we're stuck and won't be able to receive any new updates from you.

Are you aware of this issue?
Do you have any workarounds? 

Thanks

**Comment by khoi** [comment]
@bnickel can u guys at least acknowledge this?

**Comment by bnickel** [comment]
This issue is fixed in 0.3.1.  The problem was that the HeapSwiftCoreInterfaces binary target was zipped without preserving symlinks.  This caused the HeapSwiftCoreInterfaces binary be embedded multiple times, which Xcode rejected during signing.

**Comment by AndrewVebster** [comment]
Thank you for fixing the issue so quickly! We really appreciate it! 🙌
